SwissDataScienceCenter/renku-ui
# fix projects overview tabs

This PR's primary goal is to fix the project overview stats tab (see #1074).
It shows incorrect data when:
1. the project has been recently updated.
2. the user has no specific permission for a public or internal project.

Point 1 is fixed. It happened because we never refreshed the stats. In the past, we also observed that the GitLab API was occasionally returning outdated values for a while after pushing new commits. This [should be fixed now](https://gitlab.com/gitlab-org/gitlab-foss/-/merge_requests/26854).

Point 2 is partially fixed. Sadly, the GitLab project API doesn't return statistics for a public or internal project if the user doesn't have specific permissions for that. This is unexpected because it's undocumented -- I still hope [there is another way](https://gitlab.com/gitlab-org/gitlab/-/issues/296656).  
We don't show the missing stats anymore when unavailable, but we provide extra links to the GitLab UI where some of those can be inspected.

Mind that this PR includes a few other minor changes for the overview tabs:
* Avoid fetching readme on project overview sub-tabs
* Fetch commits when opening overview/commits
* Move overview/commits components to the new ProjectsOverview files
* Merge stats into a single table (it anticipates a tiny bit of #1114 for convenience)
* Move the statistics information to the project coordinator (better to anticipate steps from #1183 whenever possible)

**Preview**: https://lorenzotest.dev.renku.ch/

**How to test**
Keep an instance of https://dev.renku.ch/ and compare the following:
* Try to move around in the project overview sub-pages on the left navigation pills. You shouldn't see queries to the readme file at each click.
* Open the project overview commits for a project and wait until the full list is fetched. Then manually navigate to another project from the projects page. If you open the commits, you should see they are now correct while wrong on dev.
* Open one of your project overview stats. Then manually navigate to a public project where you have no permission. The stats will be wrong on dev, showing the previous values.
* 
Mind that the last 2 problems may not happen if you first navigate around the project, triggering fetches from GitLab for other reasons.

fix #1074
